src/test/java/io/jenkins/plugins/analysis/warnings/FlexiblePublishITest.java

```diff
@@ -0,0 +1,143 @@
+        checkDetailsViewForIssues(project, analysisResult, toolId, 2);
+        checkDetailsViewForIssues(project, analysisResult, toolId2, 4);
+    }
+
+    /**
+     * Test that the same tool can be used twice with different configuration and is aggregated afterwards.
+     */
+    @Test
+    public void shouldAnalyseJavaWithAggregation() {
+        FreeStyleProject project = createFreeStyleProject();
+        copySingleFileToWorkspace(project, JAVA_FILE, JAVA_FILE);
+        copySingleFileToWorkspace(project, JAVA_FILE2, JAVA_FILE2);
+
+        final String toolId = "hm-edu1";
+        final String toolId2 = "hm-edu2";
+
+        IssuesRecorder publisher = constructJavaIssuesRecorder(JAVA_FILE, toolId, true);
+        IssuesRecorder publisher2 = constructJavaIssuesRecorder(JAVA_PATTERN, toolId2, true);
+
+        project.getPublishersList().add(new FlexiblePublisher(Arrays.asList(
+                constructConditionalPublisher(publisher),
+                constructConditionalPublisher(publisher2)
+        )));
+
+        AnalysisResult analysisResult = scheduleSuccessfulBuild(project);
+        assertThat(analysisResult.isSuccessful()).isEqualTo(true);
+        //is there a bug with the aggregation?
+        checkDetailsViewForIssues(project, analysisResult, toolId, 6);
+    }
+
+    private IssuesRecorder constructJavaIssuesRecorder(final String patter, final String id,
+            final boolean setAggregation) {
+        return constructJavaIssuesRecorder(patter, id, setAggregation, 0, 0);
+    }
+
+    private IssuesRecorder constructJavaIssuesRecorder(final String patter, final String id,
+            final boolean setAggregation,
+            final int healthy, final int unhealthy) {
+        Java java = new Java();
+        java.setPattern(patter);
+        java.setId(id);
+        IssuesRecorder publisher = new IssuesRecorder();
+        publisher.setTools(java);
+        if (healthy != 0 && unhealthy != 0) {
+            publisher.setHealthy(healthy);
+            publisher.setUnhealthy(unhealthy);
+        }
+        publisher.setBlameDisabled(true);
+        publisher.setAggregatingResults(setAggregation);
+
+        return publisher;
+    }
+
+    private ConditionalPublisher constructConditionalPublisher(final BuildStep publisher) {
+        return new ConditionalPublisher(
+                new AlwaysRun(),
+                Collections.singletonList(
+                        publisher
+                ),
+                new Run(),
+                false,
+                null,
+                null,
+                null
+        );
+    }
+
+    private void checkDetailsViewForIssues(final Item project, final AnalysisResult analysisResult,
+            final String publisherId, final int warnings) {
+        HtmlPage detailsPage = getDetailsWebPage(project, analysisResult, publisherId);
+        IssuesTable issuesTable = new IssuesTable(detailsPage);
+        List<IssueRow> issuesTableRows = issuesTable.getRows();
```

Comment:
Since today, we are waiting for the ajax call to respond.

(And if he isn't killed, he is still waiting...)

@uhafner do you have a quick solution?

Reply:
Yes, with the design of the new page objects, the issues table needs to be created using the details tab:
```
        HtmlPage detailsPage = getDetailsWebPage(project, analysisResult, publisherId);
        DetailsTab detailsTab = new DetailsTab(detailsPage);
        IssuesTable issuesTable = detailsTab.select(TabType.ISSUES);
        List<IssueRow> issuesTableRows = issuesTable.getRows();

        assertThat(issuesTableRows.size()).isEqualTo(warnings);
```
Otherwise the tab for the issues is not selected and therefore the Ajax request is never initiated.